src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,201 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.CurrentLoginIsExists;
+import core.basesyntax.model.InvalidInputData;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static final String DEFAULT_LOGIN = "programator12341";
+    private static final String DEFAULT_PASSWORD = "passatic123";
+    private static final int DEFAULT_AGE = 19;
+    private static final int AGE_LIMIT = 18;
+    private static final int LOGIN_MIN_LENGTH = 14;
+    private static final int PASSWORD_MIN_LENGTH = 10;
+    private static final String PATTERN_FOR_PASSWORD_LOGIN = "[A-Za-z0-9]*";
+    private static final String SHORT_PASSWORD = "short";
+    private static final String SHORT_LOGIN = "short";
+    private static final String MISMATCH_LOGIN = "--------------";
+    private static final String MISMATCH_PASSWORD = "%$#%@**(*&*(";
+
+    private final RegistrationServiceImpl registrationService = new RegistrationServiceImpl();
```

Comment:
```suggestion
    private final RegistrationService registrationService = new RegistrationServiceImpl();
```
[specify an interface as a type](https://stackoverflow.com/questions/2586389/when-should-i-use-an-interface-in-java)